script.js

```diff
@@ -185,21 +185,61 @@ function fetchAllProject() {
 window.onload = () => {
   fetchAllProject();
 };
-function validateForm(event) {
-  event.preventDefault(); // Prevent form submission
+const nameInput = document.getElementById('name');
+const emailInput = document.getElementById('email');
+const messageInput = document.getElementById('message');
 
-  const emailInput = document.getElementById('email');
-  const errorMessage = document.getElementById('errorMessage');
+function loadLocalStorage() {
+  const formData = JSON.parse(window.localStorage.getItem('formData'));
+  if (formData) {
+    nameInput.value = formData.name;
+    emailInput.value = formData.email;
+    messageInput.value = formData.message;
+  }
+}
+
+function onChange(e) {
+  let formData = JSON.parse(localStorage.getItem('formData'));
+  if (!formData) {
+    formData = {};
+  }
+  const m = e.target.name;
+  formData[m] = e.target.value;
+  formData = JSON.stringify(formData);
+  window.localStorage.setItem('formData', formData);
+}
+
+nameInput.addEventListener('change', onChange);
+emailInput.addEventListener('change', onChange);
+messageInput.addEventListener('change', onChange);
+
+window.onload = () => {
+  fetchAllProject();
+  loadLocalStorage();
+};
 
-  const email = emailInput.value;
-  const lowerCaseEmail = email.toLowerCase();
+function onSubmit(e) {
+  const inputEmail = document.getElementById('email');
+  const formInfo = document.getElementById('form-info');
+  const email = inputEmail.value;
 
-  if (email !== lowerCaseEmail) {
-    errorMessage.innerText = 'Email must be in lowercase';
-    errorMessage.style.display = 'block';
-    return;
+  if (email !== email.toLowerCase()) {
+    e.preventDefault();
+    inputEmail.classList.add('invalid');
+    formInfo.classList.add('error');
+    formInfo.innerText = 'Error form is not sent! The Email should be in lower case!!';
+  } else {
+    inputEmail.classList.remove('invalid');
+    formInfo.classList.remove('error');
   }
-  document.getElementById('contact-form').submit();
 }
+const contactForm = document.getElementById('contact-form');
+contactForm.addEventListener('submit', onSubmit);
```

Comment:
- [ ] [Optional] Well done on addressing the previous reviewer's feedback! 👍 Regarding the local storage, it is indeed important to clear it after a form is successfully submitted. This ensures that the stored data does not persist unnecessarily and avoids any potential conflicts or confusion. To implement this, you can add a step in your code where, upon successful form submission, you remove or clear the relevant data from the local storage. This can be achieved using JavaScript's `localStorage.removeItem()` or `localStorage.clear()` methods. 🎉👏


[Uzair-s-Portfolio.webm](https://github.com/Uzair-Manzoor/Portfolio/assets/55840999/273cf047-fa93-4aec-946a-52135432196b)

Reply:
@shahadat3669, thanks a lot for the review and feedback. I'm on to implementing the suggested changes